src/components/Graph/Graph.tsx

```diff
@@ -103,12 +95,14 @@ const Graph: FC<GraphProps> = ({
     });
   }
 
-  const initialExpandedState = useMemo(() => {
-    return Object.keys(layout).reduce((acc: { [key: string]: boolean }, curr: string) => {
+  const layoutKeys = Object.keys(layout);
+  const initialExpandedState = layoutKeys.reduce(
```

Comment:
I'm honestly blanking on why I made this change.  I think it saves a render due to layout object changing but values being the same, causing useMemo to rerender the graph.

Reply:
Interesting.  I wonder if any reductions in rerenders are worth the loss of memoization (i.e. more rerenders that are faster overall, versus fewer rerenders that are slower overall).

Because Graph has so much state and does so much I wouldn't be surprised if the rerenders are more expensive.  It might make more sense to memoize if broken down a bit more, but that's a thought for another day.

Reply:
@KeeyanGhoreshi and I discussed and tested out some alternatives here together, but right now we can't reasonably memoize this without some greater refactoring because the dependency is only shallowly compared in React.